sqlglot/builder.py

```diff
@@ -0,0 +1,74 @@
+from functools import wraps
+
+import sqlglot.expressions as exp
+
+
+def builder(func):
+    @wraps(func)
+    def builder_method(self, *args, **kwargs):
+        query = Query()
+        query.node = self.node.copy()
```

Comment:
If we simply copy, do the new nodes still maintain a reference to the old parents?